main.tf

```diff
@@ -300,10 +259,13 @@ module "eks" {
   workers_role_name                              = var.workers_iam_role_name
   manage_cluster_iam_resources                   = var.cluster_iam_role_name == null ? true : false
   cluster_iam_role_name                          = var.cluster_iam_role_name
+  worker_create_security_group                   = false
+  worker_security_group_id                       = local.workers_security_group_id
+  cluster_create_security_group                  = false
+  cluster_security_group_id                      = local.cluster_security_group_id
```

Comment:
So in the past this code was not included and the worker and cluster security group could be created. Not as it stands this is never the case and the `worker_security_group_id` and `cluster_security_group_id` could be null. So the cluster can be created without a security group for it's workers and cluster. If you could clarify the logic here.

Reply:
correct. The module terraform-aws-modules/eks/aws creates those security groups for you, if you set those "create" flags.
I chose to always bring our own security groups. It fits most closely the pattern for all the other BYO resources: we either have a "data" item, or a "resource" item. Coding seemed slightly more straight-forward to me (it happens to also remove the need for some explicit "depends_on" items)

Reply:
What's the impact here? Folks don't always and will probably not bring their own security groups so you're changing the overall functionality and flow here. It does fit for those who do BYO SGs but not others. What's the plan there?

Reply:
I'm not sure I see a change in functionality. If folks bring their own SGs, we use then, if they don't, we create them.

The only change here is that instead of letting the eks module create the security groups, we create them ourselves (e.g. https://github.com/sassoftware/viya4-iac-aws/blob/byo_sg/security.tf#L70).